frontend/pages/tickets/[[...slug]].tsx

```diff
@@ -85,126 +53,142 @@ const Text = styled.h1`
   font-size: 1rem;
   margin: 0.5rem 0rem;
 `
-// TODO: Add auth handling gracefully.
 
-const Ticket = ({ tickets, buyers, event, home }): ReactElement => {
-  // const Ticket = ({ event }): ReactElement => {
+type TicketProps = InferGetServerSidePropsType<typeof getServerSideProps>
 
+// TODO: Add auth handling gracefully.
+
+const Ticket: React.FC<TicketProps> = ({
+  tickets,
+  buyers,
+  event,
+  home,
+}): ReactElement => {
   if (home) {
     return (
       <Center>
         Welcome to Ticketing! Please browse events with available tickets{' '}
         <a href="/events">here</a>.
       </Center>
     )
-  } else if (!tickets.totals) {
+  } else if (!tickets || !tickets.totals || !tickets.available) {
     return <Center>No tickets found with given user permissions.</Center>
   }
   const { totals, available } = tickets
-  const ticks = {}
-
-  // totals.forEach((tick) => {
-  //   if (ticks[tick.type] == null) {
-  //     ticks[tick.type] = { type: tick.type }
-  //   }
-  //   ticks[tick.type].total = tick.count
-  // })
-
-  // available.forEach((tick) => {
-  //   if (ticks[tick.type] == null) {
-  //     ticks[tick.type] = { type: tick.type }
-  //   }
-  //   ticks[tick.type].available = tick.count
-  // })
-
-  // buyers.forEach((tick) => {
-  //   if (ticks[tick.type] == null) {
-  //     ticks[tick.type] = { type: tick.type }
-  //   }
-  //   if (ticks[tick.type].buyers == null) {
-  //     ticks[tick.type].buyers = []
-  //   }
-
-  //   ticks[tick.type].buyers.push(tick.fullname)
-  // })
-
-  // tickets = []
-  // for (const [_, value] of Object.entries(ticks)) {
-  //   tickets.push(value)
-  // }
+  const tickTypes = {}
+
+  // For given ticket type, get the total and available tickets, as well as buyers
+  for (const ticket of totals) {
+    tickTypes[ticket.type] = {
+      type: ticket.type,
+      total: ticket.count,
+      price: ticket.price,
+      available: 0,
+      buyers: [],
+    }
+  }
+
+  for (const ticket of available) {
+    tickTypes[ticket.type].available = ticket.count
+  }
+
+  // Public should be able to access info about general tickets metrics, not specific buyers.
+  if (buyers.buyers) {
+    for (const buyer of buyers.buyers) {
+      tickTypes[buyer.type].buyers.push(buyer.fullname)
+    }
+  }
 
   return (
     <>
       <Container>
         <Title>All Tickets for {event.name}</Title>
-        {tickets.map((ticket, i) => (
-          <TicketCard key={i} ticket={ticket} />
+        {Object.values(tickTypes).map((ticket, i) => (
+          <TicketCard key={i} ticket={ticket} buyersPerm={buyers.buyers} />
         ))}
       </Container>
     </>
   )
 }
 
-const TicketCard = ({ ticket }) => {
+const TicketCard = ({ ticket, buyersPerm }) => {
```

Comment:
What's the type of `buyersPerm`?

Reply:
It's a list of buyer objects returned from the relevant endpoint: was using it being null or not as a stand-in for whether or not user has perms (owns the club etc.) to view ticket holders, but can change the prop to be the result of the null-check to make things cleaner.